mscore/musescore.cpp

```diff
@@ -2224,8 +2224,36 @@ static bool processNonGui()
                         }
                   }
             else if (fn.endsWith(".png")) {
-                  cs->switchToPageMode();
-                  rv = mscore->savePng(cs, fn);
+                  if (!exportScoreParts)
+                        return mscore->savePng(cs, fn);
+                  else {
+                        if (cs->excerpts().size() == 0) {
+                              QList<Excerpt*> exceprts = Excerpt::createAllExcerpt(cs);
+                      
+                              foreach(Excerpt* e, exceprts) {
```

Comment:
please use `for (Excerpt* e : excerpts) {` instead
(Yes I know the other sections in  that file set a bad example, you might as well fix them in the due course)